contracts/peripheral/YieldDirectorV2.sol

```diff
@@ -0,0 +1,451 @@
+        }
+
+        return _getOutstandingYield(principalTotal, agnosticTotal);
+    }
+
+    /************************
+     * Recipient Functions
+     ************************/
+
+    function redeemableBalance(uint256 id_) public view returns (uint256) {
+        DepositInfo storage currDeposit = depositInfo[id_];
+        return _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount);
+    }
+
+    /**
+        @notice Get redeemable sOHM balance of a recipient address
+        @param recipient_ Address of user receiving donated yield
+     */
+    function totalRedeemableBalance(address recipient_) public view returns (uint256) {
+        uint256[] storage receiptIds = recipientIds[recipient_];
+        uint256 agnosticRedeemable = 0;
+
+        for (uint256 index = 0; index < receiptIds.length; index++) {
+            DepositInfo storage currDeposit = depositInfo[receiptIds[index]];
+            agnosticRedeemable += _getOutstandingYield(currDeposit.principalAmount, currDeposit.agnosticAmount);
+        }
+
+        return agnosticRedeemable;
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index from one donor as gOHM
+        @param id_ Deposit id for this donation
+    */
+    function redeemYield(uint256 id_) external returns (uint256) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+        require(recipientLookup[id_] == msg.sender);
+
+        uint256 amountRedeemed = _redeemYield(id_);
+        if (amountRedeemed == 0) revert YieldDirector_NoRedeemableBalance();
+
+        IERC20(gOHM).safeTransfer(msg.sender, amountRedeemed);
+
+        emit Redeemed(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's donated amount of sOHM at current index
+        @param id_ Deposit id for this donation
+    */
+    function redeemYieldAsSohm(uint256 id_) external returns (uint256) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+        require(recipientLookup[id_] == msg.sender);
+
+        uint256 amountRedeemed = _redeemYield(id_);
+        if (amountRedeemed == 0) revert YieldDirector_NoRedeemableBalance();
+
+        IERC20(sOHM).approve(address(staking), amountRedeemed);
+        staking.unwrap(msg.sender, amountRedeemed);
+
+        emit Redeemed(msg.sender, amountRedeemed);
+    }
+
+    /**
+        @notice Redeem recipient's full donated amount of sOHM at current index as gOHM
+    */
+    function redeemAllYield() external returns (uint256) {
+        if (redeemDisabled) revert YieldDirector_RedeemsDisabled();
+
+        uint256 totalRedeemable = totalRedeemableBalance(msg.sender);
+        if (totalRedeemable == 0) revert YieldDirector_NoRedeemableBalance();
+
+        uint256[] storage receiptIds = recipientIds[msg.sender];
+        
+        for (uint256 index = 0; index < receiptIds.length; index++) {
+            _redeemYield(receiptIds[index]);
```

Comment:
Since you are already looping through the recieiptIds here i think you can do the adding up totalredeemable yield here. 

before the for loop
uint256 totalRedeemable = 0
inside for loop
totalRedeemable += _redeemYield(receiptIds[index]);

This way we don't have to call totalRedeemableBalance which loops through that array again.

Reply:
good call